docs/reference/tab-widgets/troubleshooting/data/restore-from-snapshot.asciidoc

```diff
@@ -0,0 +1,213 @@
++
+. When the process is complete, you will see in the **Restore Status** tab the result of your restore.
++
+[role="screenshot"]
+image::images/troubleshooting/data/kibana-restore-snapshot-result.png[{kib} select resources to restore,align="center"]
++
+For more guidance on creating and restoring snapshots see
+<<snapshot-restore, this guide>>.
+
+//end::kibana-api-ex[]
+// end::kibana[]
+
+// tag::api[]
+In order to restore the indices that are missing shards:
+
+. View the affected indices using the <<cat-indices,cat indices API>>.
++
+[source,console]
+----
+GET _cat/indices?v&health=red&h=index,status,health
+----
++
+The response will look like this:
++
+[source,console-result]
+----
+index                                status health
+.ds-my-data-stream-2022.06.17-000001 open   red
+kibana_sample_data_flights           open   red
+----
+// TEST[skip:illustration purposes only]
++
+The `red` health of the indices above indicates that these indices are missing primary shards,
+meaning they are missing data.
++
+. In order to restore the data we need to find a snapshot that contains these two indices. To find
+such a snapshot use the
+<<get-snapshot-api,get snapshot API>>.
++
+[source,console]
+----
+GET _snapshot/my_repository/*?verbose=false
+----
+// TEST[skip:illustration purposes only]
++
+The response will look like this:
++
+[source,console-result]
+----
+{
+  "snapshots" : [
+    {
+      "snapshot" : "snapshot-20200617",                                     <1>
+      "uuid" : "dZyPs1HyTwS-cnKdH08EPg",
+      "repository" : "my_repository",                                           <2>
+      "indices" : [                                                         <3>
+        ".apm-agent-configuration",
+        ".apm-custom-link",
+        ".ds-ilm-history-5-2022.06.17-000001",
+        ".ds-my-data-stream-2022.06.17-000001",
+        ".geoip_databases",
+        ".kibana-event-log-8.2.2-000001",
+        ".kibana_8.2.2_001",
+        ".kibana_task_manager_8.2.2_001",
+        "kibana_sample_data_ecommerce",
+        "kibana_sample_data_flights",
+        "kibana_sample_data_logs"
+      ],
+      "data_streams" : [ ],
+      "state" : "SUCCESS"                                                     <4>
+    }
+  ],
+  "total" : 1,
+  "remaining" : 0
+}
+----
+// TEST[skip:illustration purposes only]
++
+<1> The name of the snapshot.
++
+<2> The repository of the snapshot.
++
+<3> The indices backed up in the snapshot.
++
+<4> If the snapshot was successful.
+
+. The snapshot `snapshot-20200617` contains the two indices we want to restore.
+You might have multiple snapshots from which you could restore the target indices. Choose the latest snapshot.
++
+. Now that we found a snapshot, we will close the target indices via the use the <<indices-close, close indices API>>.
++
+[source,console]
+----
+POST kibana_sample_data_flights,.ds-my-data-stream-2022.06.17-000001/_close
```

Comment:
For the last generation backing index of a data stream, in case the data stream has more than one generation, this call will fail. 
If the last generation of a data stream needs to be restored I think we should recommend rolling over the data stream and closing the (now) N-1 generation index.

Reply:
Great catch @andreidan , I will adjust this.

Reply:
Hm, I tried it and it didn't fail, this is what I tried:

1. Created a data stream
2. Indexed 3 documents
3. Manually rolled over
4. Index one more document, confirmed that it got indexed in the last generation
5. Closed the last generation
6. Tried to index another document and I got an `index_closed_exception`.

This is the scenario that you expected to fail, right?

That said, I do think it's a nice idea to rollover because that will ensure smooth indexing even while restoring a snapshot. My only concern is that I do not want to confuse the reader too much with data stream terminology (especially if we provide this as the only troubleshooting scenario for cloud and self managed).

What do you think if we say that if the index name has the pattern `.ds-my-data-stream-xxxxx` the user can assume it's a data stream and then always rollover first. This way they do not have to check if it's the latest one. Rolling over prematurely is not the best but I think since this is an "emergency" scenario it would pay off to cut that corner. 

WDYT @andreidan & @dakrone ?

Reply:
Done: https://github.com/elastic/elasticsearch/issues/88837